packages/gatsby/src/state-machines/query-running/index.ts

```diff
@@ -14,6 +14,9 @@ export const queryStates: MachineConfig<IQueryRunningContext, any, any> = {
     SOURCE_FILE_CHANGED: {
       target: `extractingQueries`,
     },
+    ADD_NODE_MUTATION: {
+      actions: [`markNodesDirty`, `callApi`],
+    },
```

Comment:
Unfortunely this won't do what you expect. As this is a child machine, the action is run against the context of the child machine, so marking nodes as dirty won't do anything.